src/components/modules/ui.ts

```diff
@@ -691,30 +691,30 @@ export default class UI extends Module<UINodes> {
       return;
     }
 
-    if (!this.Editor.BlockManager.currentBlock) {
-      stopPropagation();
+    const isClickedBottom = event.target instanceof Element && event.target.isEqualNode(this.nodes.redactor);
 
-      this.Editor.BlockManager.insert();
-    }
+    if (isClickedBottom) {
+      stopPropagation();
 
-    /**
-     * Show the Plus Button if:
-     * - Block is an default-block (Text)
-     * - Block is empty
-     */
-    const isDefaultBlock = this.Editor.Tools.isDefault(this.Editor.BlockManager.currentBlock.tool);
+      const { BlockManager, Caret, Toolbar, Tools } = this.Editor;
 
-    if (isDefaultBlock) {
-      stopPropagation();
+      /**
+       * Insert a default-block at the bottom if:
+       * - last-block is not a default-block (Text)
+       *   to prevent unnecessary tree-walking on Tools with many nodes (for ex. Table)
+       * - Or, default-block is not empty
+       */
+      if (!Tools.isDefault(BlockManager.lastBlock.tool) || !BlockManager.lastBlock.isEmpty) {
```

Comment:
Please pull next branch and use `BlockManager.lastBlock.tool.isDefault`

Reply:
Fixed for version 2.20.0! 🎉